### PR TITLE
Microchip XEC Clock Control updates

### PR DESCRIPTION
**Pull request by jvasanth1**
update Microchip MEC172x clock control driver to support all modes. Add support for external 32KHZ_IN pin as a clock source including PINTRL to switch the GPIO to 32KHZ_IN function. Add device tree option to disable internal silicon oscillator if it is not required by the configuration. Add device tree tuning options based on crystal and board layout. Add clock control driver support for MEC15xx as well.

### CODE REVIEW

**Comment by jvasanth1** [comment]
@albertofloyd can you help to review this PR?

**Comment by jvasanth1** [comment]
@MaureenHelm can you help to review this PR?

File: samples/drivers/clock_control_xec/boards/mec1501modular_assy6885.overlay [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	internal-osc-disable;
+
+	/* debug only, not using 32KHZ_IN pin */
+	pinctrl-0 = <&tst_clk_out_gpio060 &clk_32khz_out_gpio221>;
+
+	/* 32KHZ_IN is a clock source and no debug */
+	/* pinctrl-0 = <&clk_32khz_in_gpio165>; */
+
+	/* 32KHZ_IN is a clock source with debug */
+	/* pinctrl-0 = <&clk_32khz_in_gpio165
```
Comment (albertofloyd) [inline samples/drivers/clock_control_xec/boards/mec1501modular_assy6885.overlay:25]: @jvasanth1 how does this work by having multiple redefinitions here?  Are these all the possible options for debug pins
Or is the idea that when PCR HW block is enabled all these pins will always be in alternate function?

Comment (jvasanth1) [reply samples/drivers/clock_control_xec/boards/mec1501modular_assy6885.overlay:25]: Hi Alberto, the lines 22 and 25-27 is not redefinition but commented code to list various options.

Comment (albertofloyd) [reply samples/drivers/clock_control_xec/boards/mec1501modular_assy6885.overlay:25]: understood.

**Review by albertofloyd — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**